docling/datamodel/document.py

```diff
@@ -370,4 +374,7 @@ def _detect_html_xhtml(
         if re.match(r"<!doctype\s+html|<html|<head|<body", content_str):
             return "text/html"
 
+        if re.match("<!doctype [^>]+>", content_str):
+            return "application/xml"
```

Comment:
Just if we want to be very strict and ensure that the doctype follows as a tag (requirement in XML), you could use:
```python
p = re.compile(r"<!doctype\s+(?P<root>[a-zA-Z_:][a-zA-Z0-9_:.-]*)\s+.*>\s*<(?P=root)\b")
if p.search(content_str):
    return "application/xml"  
```